R/SSmohnsrho.R

```diff
@@ -1,135 +1,183 @@
-    # This rho value is then scaled according to the number of model years
-    # for comparison between the one year and all year calculation
-    # Rho <- Rho / Number of Years
-    for (i in 1:(N - 1)) {
-      ind <- which(summaryoutput[["SpawnBio"]][["Yr"]] == startyr):which(summaryoutput[["SpawnBio"]][["Yr"]] == endyrvec[i + 1])
-      mohnSSB.all[i] <-
-        sum((summaryoutput[["SpawnBio"]][ind, i + 1] - summaryoutput[["SpawnBio"]][ind, 1]) /
-          summaryoutput[["SpawnBio"]][ind, 1]) / length(ind)
-      ind <- which(summaryoutput[["recruits"]][["Yr"]] == startyr):which(summaryoutput[["recruits"]][["Yr"]] == endyrvec[i + 1])
-      mohnRec.all[i] <-
-        sum((summaryoutput[["recruits"]][ind, i + 1] - summaryoutput[["recruits"]][ind, 1]) /
-          summaryoutput[["recruits"]][ind, 1]) / length(ind)
-      if (length(which(summaryoutput[["Bratio"]][["Yr"]] == startyr + 1)) != 0) {
-        ind <- which(summaryoutput[["Bratio"]][["Yr"]] == startyr + 1):which(summaryoutput[["Bratio"]][["Yr"]] == endyrvec[i + 1])
-        mohnBratio.all[i] <-
-          sum((summaryoutput[["Bratio"]][ind, i + 1] - summaryoutput[["Bratio"]][ind, 1]) /
-            summaryoutput[["Bratio"]][ind, 1]) / length(ind)
-      } else {
-        warning("Skipping Wood's Hole Mohns Rho on Bratio, as Bratio is not available for year after the first model year.")
-        mohnBratio.all[i] <- NA
-      }
-      if (length(which(summaryoutput[["Fvalue"]][["Yr"]] == startyr)) != 0) {
-        ind <- which(summaryoutput[["Fvalue"]][["Yr"]] == startyr):which(summaryoutput[["Fvalue"]][["Yr"]] == endyrvec[i + 1])
-        mohnF.all[i] <-
-          sum((summaryoutput[["Fvalue"]][ind, i + 1] - summaryoutput[["Fvalue"]][ind, 1]) /
-            summaryoutput[["Fvalue"]][ind, 1]) / length(ind)
-      } else {
-        warning("Skipping Wood's Hole Mohn's Rho on Fvalue, ecause Fvalue is not available for first model year.")
-        mohnF.all[i] <- NA
-      }
-    }
+  mohn.out <- list()
+  mohn.out[["SSB"]] <- sum(mohnSSB)
+  mohn.out[["Rec"]] <- sum(mohnRec)
+  mohn.out[["Bratio"]] <- sum(mohnBratio)
+  mohn.out[["F"]] <- sum(mohnF)
 
-    mohn.out <- list()
-    mohn.out[["SSB"]] <- sum(mohnSSB)
-    mohn.out[["Rec"]] <- sum(mohnRec)
-    mohn.out[["Bratio"]] <- sum(mohnBratio)
-    mohn.out[["F"]] <- sum(mohnF)
+  mohn.out[["WoodHole_SSB.all"]] <- sum(mohnSSB.all)
+  mohn.out[["WoodHole_Rec.all"]] <- sum(mohnRec.all)
+  mohn.out[["WoodHole_Bratio.all"]] <- sum(mohnBratio.all)
+  mohn.out[["WoodHole_F.all"]] <- sum(mohnF.all)
 
-    mohn.out[["WoodHole_SSB.all"]] <- sum(mohnSSB.all)
-    mohn.out[["WoodHole_Rec.all"]] <- sum(mohnRec.all)
-    mohn.out[["WoodHole_Bratio.all"]] <- sum(mohnBratio.all)
-    mohn.out[["WoodHole_F.all"]] <- sum(mohnF.all)
+  # Alaska Fisheries Science Center and Hurtado-Ferro et al. (2015) Mohn's rho
+  # https://www.afsc.noaa.gov/REFM/stocks/Plan_Team/2013/Sept/Retrospectives_2013_final3.pdf
```

Comment:
I am noting here that this link is broken and I have already sent an email to try and get an updated link that we can included in the roxygen documentation using `\href{new URL}{Alaska Fishery Science Center}`.

Reply:
Thank you. I saw it was broken but couldn't be bothered to find a new link.